docs/index.rst

```diff
@@ -3,6 +3,15 @@
     You can adapt this file completely to your liking, but it should at least
     contain the root `toctree` directive.
 
+.. toctree::
```

Comment:
What prompted you to move this to the top? It's usually kept at the bottom because it's rarely updated.

Reply:
I saw that's where it was on setuptools and pip:

- https://raw.githubusercontent.com/pypa/setuptools/main/docs/index.rst
- https://raw.githubusercontent.com/pypa/pip/21.1.3/docs/html/index.md

And I thought it made sense to keep it distinct from the code of conduct and the link URLs at the bottom.

Reply:
Ah, I see. Well. These ones probably have it there because of the short index documents but when it's long, it's usually annoying to scroll through it. OTOH it'd be nice for twine to reuse parts of the readme. This would make its index document source short too. I've been meaning to prepare a PR with that.

Reply:
Here it is: https://github.com/pypa/twine/pull/784.